lib/api.mjs

```diff
@@ -93,6 +79,54 @@ async function init() {
   }
   app.use(session(sessionOpts));
 
+  async function getUserSessionForAuthToken(accessToken, req) {
+    const profReq = await global.fetch(`https://www.googleapis.com/oauth2/v1/tokeninfo?access_token=${accessToken}`, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+    const profile = await profReq.json();
+    // Check if an entry already exists for this user:
+    const results = await db('users')
+      .select('id')
+      .where('users.account_type', GOOGLE_ACCOUNT_TYPE)
+      .where('users.email', profile.email);
+    if (!results.length) {
+      const { id } = (await db('users').insert({
+        email: profile.email,
+        account_type: GOOGLE_ACCOUNT_TYPE,
+      }, ['id']))[0];
+      req.session.userId = id;
+    } else {
+      const { id } = results[0];
+      req.session.userId = id;
+    }
+  }
+
+  const helpers = {
+    async authenticated(req, res, next) {
+      req.userId = req.session.userId;
+
+      // TODO FIXME: this will fetch session for every request that only has auth token, i.e. every
+      // mobile request - should be cached in e.g. redis
+      if (!req.userId && req.headers.authorization) {
+        const authToken = req.headers.authorization.substring('Bearer '.length);
+        await getUserSessionForAuthToken(authToken, req);
+        req.userId = req.session.userId;
+      }
```

Comment:
this is the only really new code - it always pings google for a session, so that's not ideal